packages/client/bin/cli.ts

```diff
@@ -431,6 +438,46 @@ async function startClient(config: Config, customGenesisState?: GenesisState) {
     // Regular client start
     await client.start()
   }
+
+  if (args.loadBlocksFromRlp !== undefined) {
+    // Specifically for Hive simulator, preload a first block provided in RLP format
+    const blockRlp = readFileSync(args.loadBlocksFromRlp)
+    const blocks: Block[] = []
+    let buf = RLP.decode(blockRlp, true)
+    let done = false
+    while (!done) {
+      try {
+        const block = Block.fromValuesArray(arrToBufArr(buf.data) as unknown as BlockBuffer, {
+          common: config.chainCommon,
+          hardforkByBlockNumber: true,
+        })
+        blocks.push(block)
+        buf = RLP.decode(buf.remainder, true)
+        config.logger.info(
+          `Preloading block 0x${short(block.header.hash().toString('hex'))} - ${
+            block.header.number
+          }`
+        )
+      } catch {
+        config.logger.info('Encountered invalid block while preloading chain')
```

Comment:
If this is the case, should we not immediately quit the client, instead of proceeding?

Reply:
https://github.com/ethereum/hive/blob/master/docs/clients.md No, we're supposed to import to to the last valid block, per the [hive docs](https://github.com/ethereum/hive/blob/master/docs/clients.md)